adapters/repos/db/migrator.go

```diff
@@ -35,13 +36,28 @@ import (
 	"github.com/weaviate/weaviate/usecases/sharding"
 )
 
+// offloadProvider is an interface has to be implemented by modules
+// to get the backend (s3, azure blob storage, google cloud storage) module
+type offloadProvider interface {
+	// OffloadBackend returns the backend module for (s3, azure blob storage, google cloud storage)
+	OffloadBackend(backend string) (modulecapabilities.OffloadCloud, error)
+}
+
 type Migrator struct {
 	db     *DB
+	cloud  modulecapabilities.OffloadCloud
 	logger logrus.FieldLogger
+	nodeId string
 }
 
-func NewMigrator(db *DB, logger logrus.FieldLogger) *Migrator {
-	return &Migrator{db: db, logger: logger}
+func NewMigrator(db *DB, logger logrus.FieldLogger, provider offloadProvider, nodeName string) (*Migrator, error) {
+	cloud, err := provider.OffloadBackend("offload-s3")
```

Comment:
does it mean that the offload provider and connectivity to S3 is now mandatory to start Weaviate?

Reply:
Na,  i just have it is a TODO, later you will find that it will return the migrator without the the cloud provider. i will resolve this TODO in a follow up PR were the real implementations are